src/DataBoxEdge/DataBoxEdge/Common/Cmdlets/Devices/DataBoxEdgeDeviceInvokeCmdletBase.cs

```diff
@@ -0,0 +1,215 @@
+        [Parameter(Mandatory = true,
+            ParameterSetName = InvokeScanForUpdateByDeviceObjectParameterSet,
+            HelpMessage = HelpMessageDevice.ScanUpdateHelpMessage)]
+        [ValidateNotNullOrEmpty]
+        public SwitchParameter ScanForUpdate { get; set; }
+
+        [Parameter(Mandatory = true,
+            ParameterSetName = InvokeInstallUpdateParameterSet,
+            HelpMessage = HelpMessageDevice.InstallUpdateHelpMessage)]
+        [Parameter(Mandatory = true,
+            ParameterSetName = InvokeInstallUpdatesByResourceIdParameterSet,
+            HelpMessage = HelpMessageDevice.InstallUpdateHelpMessage)]
+        [Parameter(Mandatory = true,
+            ParameterSetName = InvokeInstallUpdatesByDeviceObjectParameterSet,
+            HelpMessage = HelpMessageDevice.InstallUpdateHelpMessage)]
+        [ValidateNotNullOrEmpty]
+        public SwitchParameter InstallUpdate { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = Constants.AsJobHelpMessage)]
+        public SwitchParameter AsJob { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = Constants.PassThruHelpMessage)]
+        public SwitchParameter PassThru;
+
+
+        private bool ScanForUpdates()
+        {
+            DevicesOperationsExtensions.ScanForUpdates(
+                this.DataBoxEdgeManagementClient.Devices,
+                this.Name,
+                this.ResourceGroupName);
+            return true;
+        }
+
+        private bool DownloadUpdates()
+        {
+            DevicesOperationsExtensions.DownloadUpdates(
+                this.DataBoxEdgeManagementClient.Devices,
+                this.Name,
+                this.ResourceGroupName);
+            return true;
+        }
+
+        private bool InstallUpdates()
+        {
+            DevicesOperationsExtensions.InstallUpdates(
+                this.DataBoxEdgeManagementClient.Devices,
+                this.Name,
+                this.ResourceGroupName);
+            return true;
+        }
+
+        public override void ExecuteCmdlet()
+        {
+            var result = false;
+            if (this.IsParameterBound(c => c.ResourceId))
+            {
+                var resourceIdentifier = new DataBoxEdgeResourceIdentifier(this.ResourceId);
+                this.ResourceGroupName = resourceIdentifier.ResourceGroupName;
+                this.Name = resourceIdentifier.ResourceName;
+            }
+            else if (this.IsParameterBound(c => c.DeviceObject))
+            {
+                this.ResourceGroupName = DeviceObject.ResourceGroupName;
+                this.Name = DeviceObject.Name;
+            }
+
+            if (this.ScanForUpdate.IsPresent)
+            {
+                result = ScanForUpdates();
+            }
+            else if (this.FetchUpdate.IsPresent)
+            {
+                result = InstallUpdates();
+            }
+            else if (this.InstallUpdate.IsPresent)
```

Comment:
This should be `DownloadUpdates`. Do you have any test to cover this?

Reply:
> `Mandatory` means the parameter is required in the parameter set. I think many usage maybe wrong in these 5 cmdlets. Please check for this. I have point out some, but there are too many usage in the code, I cannot check them one by one.

ParameterSets are used to enforce using one SwitchParameter at a time